python/tank/hook.py

```diff
@@ -251,6 +253,64 @@ def get_publish_paths(self, sg_publish_data_list):
 
         return paths
 
+    @property
+    def disk_location(self):
+        """
+        The folder on disk where this item is located.
+        This can be useful if you want to write hook code
+        to retrieve a local resource::
+
+            hook_icon = os.path.join(self.disk_location, "icon.png")
+        """
+        path_to_this_file = os.path.abspath(sys.modules[self.__module__].__file__)
```

Comment:
Ah! This is because `__file__` will always return the base class (`hook.py`, `bundle.py`) location - but we want to return the location of the current hook or current app.
I'll add comments to clarify.